### ISSUE TITLE
save_graph cannot insert osmid, already exist

### ISSUE THREAD

**Issue by SevLo** [gboeing/osmnx#638]
Hi, I hope it's not me doing something wrong and that it's legitimate to open this report.

**Problem description**
  - What did you do?: tried to save a condoslidated graph
  - What did you expect to happen?: to find it at the lcoation specified
  - And what actually happened instead?: error was thrown:

_Traceback (most recent call last):
  File "/xyz/testscript.py", line 19, in <module>
    ox.save_graph_xml(G2, filepath='./data/piedmont.osm')
  File "/Users/Severin/miniconda3/envs/ox/lib/python3.9/site-packages/osmnx/osm_xml.py", line 192, in save_graph_xml
    gdf_nodes = gdf_nodes.reset_index().rename(columns={"osmid": "id"})
  File "/Users/Severin/miniconda3/envs/ox/lib/python3.9/site-packages/pandas/core/frame.py", line 5013, in reset_index
    new_obj.insert(0, name, level_values)
  File "/Users/Severin/miniconda3/envs/ox/lib/python3.9/site-packages/pandas/core/frame.py", line 3760, in insert
    self._mgr.insert(loc, column, value, allow_duplicates=allow_duplicates)
  File "/Users/Severin/miniconda3/envs/ox/lib/python3.9/site-packages/pandas/core/internals/managers.py", line 1191, in insert
    raise ValueError(f"cannot insert {item}, already exists")
ValueError: cannot insert osmid, already exists_

**Environment information**
  - What operating system are you using?: MacOS Big Sur
  - What Python version are you using?: 3.7
  - What OSMnx version are you using? 1.0.0
  - Provide a complete list of your environment's packages and their versions (for example, run `conda list` or `pip list` then paste the output between the two "details" tags below)

<details>
# Name                    Version                   Build  Channel
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
boost-cpp                 1.74.0               he5d75e3_0    conda-forge
branca                    0.4.2              pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py39h66d5b7b_1001    conda-forge
bzip2                     1.0.8                hc929b4f_4    conda-forge
c-ares                    1.17.1               hc929b4f_0    conda-forge
ca-certificates           2020.12.5            h033912b_0    conda-forge
cairo                     1.16.0            he8db5a5_1006    conda-forge
certifi                   2020.12.5        py39h6e9494a_1    conda-forge
cffi                      1.14.4           py39h7786acb_1    conda-forge
cfitsio                   3.470                h01dc385_7    conda-forge
chardet                   4.0.0            py39h6e9494a_0    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.1              pyhd8ed1ab_0    conda-forge
cryptography              3.3.1            py39h79a2c39_0    conda-forge
curl                      7.71.1               hcb81553_8    conda-forge
cycler                    0.10.0                     py_2    conda-forge
decorator                 4.4.2                      py_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
expat                     2.2.9                hb1e8313_2    conda-forge
fiona                     1.8.18           py39hd493767_0    conda-forge
folium                    0.12.0             pyhd8ed1ab_0    conda-forge
fontconfig                2.13.1            h186bf41_1002    conda-forge
freetype                  2.10.4               h3f75d11_0    conda-forge
freexl                    1.0.5             h0b31af3_1002    conda-forge
gdal                      3.1.4            py39h906d53f_0    conda-forge
geopandas                 0.8.1                      py_0    conda-forge
geos                      3.8.1                h4a8c4bd_0    conda-forge
geotiff                   1.6.0                hc1004f4_3    conda-forge
gettext                   0.19.8.1          h7937167_1005    conda-forge
giflib                    5.2.1                hbcb3906_2    conda-forge
glib                      2.66.4               h22858aa_1    conda-forge
hdf4                      4.2.13            h71d84a9_1004    conda-forge
hdf5                      1.10.6          nompi_hc5d9132_1113    conda-forge
icu                       67.1                 hb1e8313_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
joblib                    1.0.0              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   hbcb3906_0    conda-forge
json-c                    0.13.1            h575e443_1002    conda-forge
kealib                    1.4.14               h82515a5_0    conda-forge
kiwisolver                1.3.1            py39heea7e63_0    conda-forge
krb5                      1.17.2               h60d9502_0    conda-forge
lcms2                     2.11                 h11f7e16_1    conda-forge
libblas                   3.9.0                7_openblas    conda-forge
libcblas                  3.9.0                7_openblas    conda-forge
libcurl                   7.71.1               h9bf37e3_8    conda-forge
libcxx                    11.0.0               h4c3b8ed_1    conda-forge
libdap4                   3.20.6               h993cace_1    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.3                  h046ec9c_2    conda-forge
libgdal                   3.1.4                hd7bf8dc_0    conda-forge
libgfortran               5.0.0               h6c81a4c_15    conda-forge
libgfortran5              9.3.0               h6c81a4c_15    conda-forge
libglib                   2.66.4               h7424822_1    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
libkml                    1.3.0             h8ca2c65_1012    conda-forge
liblapack                 3.9.0                7_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h9d8a93f_107    conda-forge
libnghttp2                1.41.0               h7580e61_2    conda-forge
libopenblas               0.3.12          openmp_h54245bb_1    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libpq                     12.3                 h7049927_3    conda-forge
libspatialindex           1.9.3                h4a8c4bd_3    conda-forge
libspatialite             5.0.0                hc85de81_0    conda-forge
libssh2                   1.9.0                h8a08a2b_5    conda-forge
libtiff                   4.2.0                h355d032_0    conda-forge
libwebp-base              1.1.0                hbcb3906_3    conda-forge
libxml2                   2.9.10               h2c6e4a5_2    conda-forge
llvm-openmp               11.0.0               h73239a0_1    conda-forge
lz4-c                     1.9.3                h046ec9c_0    conda-forge
markupsafe                1.1.1            py39h66d5b7b_2    conda-forge
matplotlib-base           3.3.3            py39h5674cf4_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
networkx                  2.5                        py_0    conda-forge
numpy                     1.19.5           py39he588a01_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.3.1                h6cbf5cd_3    conda-forge
openssl                   1.1.1i               h35c211d_0    conda-forge
osmnx                     1.0.0              pyhd3deb0d_0    conda-forge
pandas                    1.2.0            py39h089d6f7_0    conda-forge
pcre                      8.44                 hb1e8313_0    conda-forge
pillow                    8.1.0            py39hc06e9b5_0    conda-forge
pip                       20.3.3             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               hbcb3906_0    conda-forge
poppler                   0.89.0               h3232a60_1    conda-forge
poppler-data              0.4.10                        0    conda-forge
postgresql                12.3                 he852019_3    conda-forge
proj                      7.1.1                h45baca5_3    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.1.post1      py39hd525a1d_3    conda-forge
pysocks                   1.7.1            py39h2c36a5b_2    conda-forge
python                    3.9.1           h1d169a7_2_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.9                      1_cp39    conda-forge
pytz                      2020.5             pyhd8ed1ab_0    conda-forge
readline                  8.0                  h0678c8f_2    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
rtree                     0.9.7            py39h7d0d40a_0    conda-forge
scikit-learn              0.24.0           py39h1f17f7f_0    conda-forge
scipy                     1.6.0            py39hca3e1c3_0    conda-forge
setuptools                49.6.0           py39h2c36a5b_2    conda-forge
shapely                   1.7.1            py39h7edaf14_1    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.34.0               h17101e1_0    conda-forge
threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
tiledb                    2.1.6                h6a9efa5_1    conda-forge
tk                        8.6.10               h0419947_1    conda-forge
tornado                   6.1              py39h5a22ff9_0    conda-forge
tzcode                    2020f                h35c211d_0    conda-forge
tzdata                    2020f                he74cb21_0    conda-forge
urllib3                   1.26.2             pyhd8ed1ab_0    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
xerces-c                  3.2.3                h0a0444a_1    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
zlib                      1.2.11            h7795811_1010    conda-forge
zstd                      1.4.8                hf387650_1    conda-forge
</details>

**Provide a complete minimal reproducible example**

```python
import networkx as nx
import osmnx as ox
import requests
import matplotlib.cm as cm
import matplotlib.colors as colors
#matplotlib inline
ox.config(log_console=True)
ox.__version__

ox.config(all_oneway=True, log_console=True, use_cache=True)

# download/model a street network for some city then visualize it
G = ox.graph_from_place('Grabenstaett, Germany', network_type='drive')

G_proj = ox.project_graph(G)

G2 = ox.consolidate_intersections(G_proj, rebuild_graph=True, tolerance=15, dead_ends=False)

ox.save_graph_geopackage(G2, filepath='./data/Grabenstaett.gpkg')
```

**Comment by gboeing** [comment]
@SevLo thanks it looks like this is a bug in how `consolidate_intersections` works with the node indexing scheme introduced in v1.0. I'll put a fix together shortly, but in the meantime, you can get your code working by running (immediately before saving the geopackage):

```python
for node, data in G2.nodes(data=True):
    if 'osmid' in data:
        data['osmid_original'] = data.pop('osmid')
```

**Comment by gboeing** [comment]
See proposed fix in #640
